modules/videoio/src/videoio_registry.cpp

```diff
@@ -333,7 +333,24 @@ class VideoBackendRegistry
 
 } // namespace
 
-namespace videoio_registry {
+namespace videoio_registry { namespace internal {
+    const VideoBackendInfo& getBackendInfo(VideoCaptureAPIs api)
+    {
+        if (api == CAP_ANY)
+            CV_Error(Error::StsError, "Wrong backend ID: CAP_ANY");
+
+        const int N = sizeof(builtin_backends)/sizeof(builtin_backends[0]);
+        for (size_t i = 0; i < N; i++)
+        {
+            const VideoBackendInfo& info = builtin_backends[i];
+            if (info.id == api)
+                return info;
+        }
+
+        CV_Error(Error::StsError, "Unknown backend ID");
+        return builtin_backends[0]; // unreachable but necessary to silence warnings
```

Comment:
> silence warnings

Which build configuration provides warning?

---

> `builtin_backends[0]`

This array may be empty. So even a zero/first element is invalid.

Reply:
> > silence warnings
> 
> Which build configuration provides warning?
I can't tell you exactly because it was the CI that was failing with a warning because of returning a temporary. Changing the code to this suppressed the warning.
> > `builtin_backends[0]`
> 
> This array may be empty. So even a zero/first element is invalid.
Of course, but it's (supposed) to be unreachable. How would you like me to handle that case?